packages/feedback/src/screenshot/components/imageEditorWrapper.tsx

```diff
@@ -0,0 +1,139 @@
+import { h } from 'preact';
+import type { ToolKey } from './useImageEditor';
+import { Tools, useImageEditor } from './useImageEditor';
+import { ArrowIcon, HandIcon, PenIcon, RectangleIcon } from './useImageEditor/icons';
+import { WINDOW } from './../../constants';
+import type { ComponentType } from 'preact';
+import { createScreenshotAnnotateStyles } from './imageEditorWrapper.css';
+
+export interface Rect {
+  height: number;
+  width: number;
+  x: number;
+  y: number;
+}
+interface ImageEditorWrapperProps {
+  onCancel: () => void;
+  onSubmit: (screenshot: HTMLCanvasElement | null) => void;
+  src: HTMLCanvasElement;
+}
+
+const iconMap: Record<ToolKey, ComponentType> = {
+  arrow: ArrowIcon,
+  pen: PenIcon,
+  rectangle: RectangleIcon,
+  select: HandIcon,
+};
+
+const getCanvasRenderSize = (canvas: HTMLCanvasElement, containerElement: HTMLDivElement) => {
+  const canvasWidth = canvas.width;
+  const canvasHeight = canvas.height;
+  const maxWidth = containerElement.getBoundingClientRect().width;
+  const maxHeight = containerElement.getBoundingClientRect().height;
+  // fit canvas to window
+  let width = canvasWidth;
+  let height = canvasHeight;
+  const canvasRatio = canvasWidth / canvasHeight;
+  const windowRatio = maxWidth / maxHeight;
+
+  if (canvasRatio > windowRatio && canvasWidth > maxWidth) {
+    height = (maxWidth / canvasWidth) * canvasHeight;
+    width = maxWidth;
+  }
+
+  if (canvasRatio < windowRatio && canvasHeight > maxHeight) {
+    width = (maxHeight / canvasHeight) * canvasWidth;
+    height = maxHeight;
+  }
+
+  return { width, height };
+};
+
+const srcToImage = (src: string): HTMLImageElement => {
+  const image = new Image();
+  image.src = src;
+  return image;
+};
+
+function ToolIcon({ tool }: { tool: ToolKey }) {
+  const Icon = tool ? iconMap[tool] : HandIcon;
+  return <Icon />;
+}
+
+export function ImageEditorWrapper({ src, onCancel, onSubmit }: ImageEditorWrapperProps) {
+  const wrapperRef = useRef<HTMLDivElement>(null);
+  const [canvas, setCanvas] = useState<HTMLCanvasElement | null>(null);
+  const styles = useMemo(() => ({ __html: createScreenshotAnnotateStyles().innerText }), []);
+
+  const resizeCanvas = useCallback(() => {
+    if (!canvas || !wrapperRef.current) {
+      return;
+    }
+    // fit canvas to window
+    const { width, height } = getCanvasRenderSize(canvas, wrapperRef.current);
+    canvas.style.width = `${width}px`;
+    canvas.style.height = `${height}px`;
+  }, [canvas]);
+
+  // const image = useMemo(() => srcToImage(src), [src]);
+  const { selectedTool, setSelectedTool, selectedColor, setSelectedColor, getBlob } = useImageEditor({
+    canvas,
+    image: src,
+    onLoad: resizeCanvas,
+  });
+
+  useEffect(() => {
+    resizeCanvas();
+    WINDOW.addEventListener('resize', resizeCanvas);
+    return () => {
+      WINDOW.removeEventListener('resize', resizeCanvas);
+    };
+  }, [resizeCanvas]);
```

Comment:
You'll want to throttle or debounce this otherwise many `resize` events end up getting fired when the window resizes and it will slowdown the browser due to the number of events it has to handle and forced reflows inside of `resizeCanvas`